kernel/task.c

```diff
@@ -94,6 +94,11 @@ void (*task_run_hook)(void) = NULL;
 
 static void *task_run(void *task) {
     current = task;
+#if __APPLE__
```

Comment:
Could you move the ifdefs into a function, maybe in sync.h, so they don't have to be duplicated for both fork and exec?